File: MyFbApp/MyFbAppFSharp/ConfigurationLoader.fs
namespace MyFbAppFSharp

open Newtonsoft.Json
open System.IO
open GalaSoft.MvvmLight.Ioc
open System
open System.Reflection

module ConfigurationLoader = 
    
    type Config() =
        member val facebookAuthUrl = "" with get, set
        member val facebookRedirectUrl = "" with get, set
        member val facebookAppId = "" with get, set
        member val facebookAppSecret = "" with get, set
        member val scope = "" with get, set
        member val FbProfileUrl = "" with get, set
        member val FbUserPostUrl = "" with get, set
        member val FbUserPostCommentsUrl = "" with get, set

    let LoadConfig (embeddedResourceStream: Stream) =
        let streamReader = new StreamReader(embeddedResourceStream)
        let jsonString = streamReader.ReadToEnd()
        let configuration = JsonConvert.DeserializeObject<Config>(jsonString)
        configuration

File: MyFbApp/MyFbAppLibrary/Configuration/ConfigurationLoader.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.IO;
using System.Reflection;
using System.Text;
using GalaSoft.MvvmLight.Ioc;
using System.Threading.Tasks;

namespace MyFbAppLib.Configuration
{
    public class ConfigurationLoader
    {
        public void LoadConfig(Stream embeddedResourceStream)
        {
            if (embeddedResourceStream == null)
                return;

            using (var streamReader = new StreamReader(embeddedResourceStream))
            {
                var jsonString = streamReader.ReadToEnd();
               
                var configuration = JsonConvert.DeserializeObject<Config>(jsonString);

                if (configuration == null)
                    return;

                SimpleIoc.Default.Register<IConfiguration>(() => configuration);
            }
        }
    }
}
